docs/arch/adr-009-storage-migration.md

```diff
@@ -0,0 +1,61 @@
+# ADR 9: OpenEBS to OpenEBS + Rook Migration
+
+## Context
+
+Single node installs should not use Rook, as doing so adds complexity, increases resource usage, and invites disk corruption.
+Multi node installs very often require distributed storage, best provided by Rook.
+
+Many customers will move from a single-node install to a multi-node install as circumstances change.
+This migration should be as simple as possible.
+Vendors generally should also not need to specify multiple installer specs for single and multi-node installs.
+
+## Decision
+
+Provide a migration path from OpenEBS to OpenEBS + Rook, triggered by the presence of 3+ nodes.
+
+## Solution
+
+A new field, "minimum node count", will be added to the rook config. (minimum value 2, requires openebs localpv to be present)
+When it is set, the rook operator will be installed on single node installs, but no ceph cluster will be created.
+
+An example spec is provided below:
+
+```yaml
+spec:
+  rook:
+    version: "1.10.x"
+    storageClassName: "distributed"
+    minimumNodeCount: 3
+  openebs:
+    version: "3.4.x"
+    isLocalPVEnabled: true
+    localPVStorageClassName: "local"
+```
+
+When installed on a single node, two storageclasses will be available - "scaling" and "local". 
+The default storage class will be "scaling", and both storage classes will be backed by openebs localpv.
```

Comment:
How is this going to work in currently installed cluster where the user has named OpenEBS storage class as `foo` and wants to have its data automatically migrated to `distributed` ?

Reply:
it'll be based on whatever openebs storageclass name you have specified. The only one that isn't configurable is `scaling`, and that will be created upon updating to this kurl version with the appropriate config.

If someone for some reason has an existing storageclass named `scaling`, I'm happy to say I'll burn that bridge when I get to it.